src/Payloads/Refund.php

```diff
@@ -0,0 +1,127 @@
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ * @author    Alma / Nabla SAS <contact@getalma.eu>
+ * @copyright Copyright (c) 2018 Alma / Nabla SAS
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ *
+ */
+
+namespace Alma\API\Payloads;
+
+class Refund
+{
+    /* @param string */
+    private $id;
+
+    /* @param int */
+    private $amount = 0;
+
+    /* @param string */
+    private $merchantReference = '';
+
+    /* @param string */
+    private $comment = '';
+
+    public function __construct($id) {
+        $this->setId($id);
+    }
+
+    /**
+     * @return string
+     */
+    public function getId() {
+        return $this->id;
+    }
+
+    /**
+     * @param string $id
+     */
+    public function setId($id) {
+        $this->id = $id;
+    }
+
+    /**
+     * @return int
+     */
+    public function getAmount() {
+        return $this->amount;
+    }
+
+    /**
+     * @param int $amount
+     */
+    public function setAmount($amount) {
+        $this->amount = $amount;
+    }
+
+    /**
+     * @return string
+     */
+    public function getMerchantReference() {
+        return $this->merchantReference;
+    }
+
+    /**
+     * @param string $merchantReference
+     */
+    public function setMerchantReference($merchantReference) {
+        $this->merchantReference = $merchantReference;
+    }
+
+    /**
+     * @return string
+     */
+    public function getComment() {
+        return $this->comment;
+    }
+
+    /**
+     * @param string $comment
+     */
+    public function setComment($comment) {
+        $this->comment = $comment;
+    }
+
+    /**
+     * @return array
+     */
+    public function getRequestBody() {
+        $requestBody = [
+            "merchant_reference" => $this->getMerchantReference(),
+            "comment" => $this->getComment(),
+        ];
+        if ($this->getAmount() > 0) {
+            $requestBody["amount"] = $this->getAmount();
+        }
+        return $requestBody;
+    }
+
+    public static function create($id, $amount = 0, $merchantReference = "", $comment = "")
+    {
+        $refundPayload = new self($id);
+        if ($amount !== 0) {
```

Comment:
I think it is not possible to refund an amount `<= 0` ... maybe we should have an Exception here or in the setter (this is the role of the `Payload` object to validate our data, alright ?)